packages/odyssey-react-mui/src/MenuItem.tsx

```diff
@@ -10,40 +10,63 @@
  * See the License for the specific language governing permissions and limitations under the License.
  */
 
-import { MenuItem as MuiMenuItem } from "@mui/material";
+import {
+  MenuItem as MuiMenuItem,
+  MenuItemProps as MuiMenuItemProps,
+} from "@mui/material";
 import { menuItemClasses } from "@mui/material/MenuItem";
-import type { MenuItemProps as MuiMenuItemProps } from "@mui/material";
-import { memo, useContext } from "react";
+import { ReactNode, memo, useCallback, useContext } from "react";
 
 import { MenuContext } from "./MenuContext";
 
-export interface MenuItemProps
-  extends Omit<
-    MuiMenuItemProps,
-    | "component"
-    | "dense"
-    | "disableGutters"
-    | "divider"
-    | "focusVisibleClassName"
-  > {
-  /**
-   * Toggles whether or not the MenuItem represents a destructive action.
-   */
+export type MenuItemProps = {
+  children: ReactNode;
+  content?: string;
+  hasInitialFocus?: boolean;
+  isSelected?: boolean;
   isDestructive?: boolean;
-}
+  onClick?: MuiMenuItemProps["onClick"];
+  rel?: string;
+  rev?: string;
+  value?: string;
+};
 
-const MenuItem = ({ isDestructive, ...props }: MenuItemProps) => {
+const MenuItem = ({
+  children,
+  content,
+  hasInitialFocus,
+  isSelected,
+  isDestructive,
+  onClick: onClickProp,
+  rel,
+  rev,
+  value,
+}: MenuItemProps) => {
   const { closeMenu } = useContext(MenuContext);
 
+  const onClick = useCallback(
+    (event: React.MouseEvent<HTMLLIElement, MouseEvent>) => {
+      onClickProp?.(event);
+      closeMenu();
+    },
+    [onClickProp, closeMenu]
+  );
+
   return (
     <MuiMenuItem
-      {...props}
-      onClick={closeMenu}
+      content={content}
+      /* eslint-disable-next-line jsx-a11y/no-autofocus */
+      autoFocus={hasInitialFocus}
```

Comment:
I'm really wondering if we should just remove the ESLint rule altogether. No one's gonna be randomly adding this prop to Odyssey components where it doesn't belong.

Thoughts?

Reply:
I'm in favor of removing it; the only time I'm even aware of it is when we're disabling it.

Reply:
Let's add a ticket to remove it from our ESLint rules and also remove the ESLint-disabling comments as well.